src/KDSoapClient/KDSoapUdpClient.cpp

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (C) 2019  Casper Meijn <casper@meijn.net>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "KDSoapUdpClient.h"
+#include "KDSoapUdpClient_p.h"
+
+#include "KDSoapMessage.h"
+#include "KDSoapMessageReader_p.h"
+#include "KDSoapMessageWriter_p.h"
+#include <QNetworkInterface>
+
+static const QHostAddress AnyIPv4(QLatin1String("0.0.0.0"));
+
+static bool isMulticastAddress(const QHostAddress &address) {
+    if (address.protocol() == QAbstractSocket::IPv4Protocol) {
+        return address.isInSubnet(QHostAddress (QLatin1String("224.0.0.0")), 4);
+    } else if (address.protocol() == QAbstractSocket::IPv6Protocol) {
+        return address.isInSubnet(QHostAddress (QLatin1String("ff00::")), 8);
+    }
+    return false;
+}
+
+KDSoapUdpClient::KDSoapUdpClient(QObject *parent) : 
+    QObject(parent),
+    d_ptr(new KDSoapUdpClientPrivate(this))
+{
+    Q_D(KDSoapUdpClient);
+    d->socketIPv4 = new QUdpSocket(this);
+    connect(d->socketIPv4, SIGNAL(readyRead()), d, SLOT(readyRead()));
+    d->socketIPv6 = new QUdpSocket(this);
+    connect(d->socketIPv6, SIGNAL(readyRead()), d, SLOT(readyRead()));
+}
+
+KDSoapUdpClient::~KDSoapUdpClient()
+{
+    delete d_ptr;
+}
+
+bool KDSoapUdpClient::bind(quint16 port, QAbstractSocket::BindMode mode) {
+    Q_D(KDSoapUdpClient);
+    bool rc = true;
+    // Workaround for lack of dual stack sockets in Qt4
+    // Qt5 supports binding to QHostAddress::Any, which will listen on both IPv4 and IPv6 interfaces.
+    rc = d->socketIPv4->bind(AnyIPv4, port, mode) && rc;
```

Comment:
I am surprised that the code was not compatible with Qt4. I did build this patch using Qt4 to make sure it works. Maybe I build an earlier version and broke the build after that.

In your commit fixing the Qt4 build you mention "once we drop Qt4 support". Are there any plans for dropping Qt4 support? In what time frame could that be expected? Would it be acceptable to make this feature Qt5 only? 

The recent PKCS12 feature only works with Qt5.4, because of the API improvements in that version. But this feature should be able to work correctly using Qt4.

Personally I don't use Qt4 and I can't easily test using Qt4 either (as you have noticed by me breaking Qt4 builds several times ;-). I would happily modify KDSoapUdpClient to be Qt5 only and remove the workaround.